builtin/providers/azure/resource_azure_hosted_service.go

```diff
@@ -0,0 +1,190 @@
+
+	log.Println("[INFO] Querying for hosted service existence.")
+	serviceName := d.Get("service_name").(string)
+	_, err := hostedServiceClient.GetHostedService(serviceName)
+	if err != nil {
+		if management.IsResourceNotFoundError(err) {
+			// it means that the hosted service has been deleted in the meantime...
+			d.SetId("")
+			return false, nil
+		} else {
+			return false, fmt.Errorf("Failed to query for hosted service name availability: %s", err)
+		}
+	}
+
+	return true, nil
+}
+
+// resourceAzureHostedServiceRead does all the necessary API calls
+// to read the state of a hosted service from Azure.
+func resourceAzureHostedServiceRead(d *schema.ResourceData, meta interface{}) error {
+	azureClient := meta.(*AzureClient)
+	hostedServiceClient := hostedservice.NewClient(azureClient.managementClient)
+
+	log.Println("[INFO] Querying for hosted service info.")
+	serviceName := d.Get("service_name").(string)
+	hostedService, err := hostedServiceClient.GetHostedService(serviceName)
+	if err != nil {
+		if management.IsResourceNotFoundError(err) {
+			// it means the hosted service was deleted in the meantime,
+			// so we must remove it here:
+			d.SetId("")
+			return nil
+		} else {
+			return fmt.Errorf("Failed to get hosted service: %s", err)
+		}
+	}
+
+	log.Println("[DEBUG] Reading hosted service query result data.")
+	d.Set("service_name", hostedService.ServiceName)
+	d.Set("url", hostedService.URL)
+	d.Set("location", hostedService.Location)
+	d.SetId(hostedService.Label)
+	d.Set("description", hostedService.Description)
+	d.Set("status", hostedService.Status)
+	d.Set("reverse_dns_fqdn", hostedService.ReverseDNSFqdn)
+	d.Set("default_certificate_thumbprint", hostedService.DefaultWinRmCertificateThumbprint)
+
+	return nil
+}
+
+// resourceAzureHostedServiceUpdate does all the necessary API calls to
+// update some settings of a hosted service on Azure.
+func resourceAzureHostedServiceUpdate(d *schema.ResourceData, meta interface{}) error {
+	return nil
+}
+
+// resourceAzureHostedServiceDelete does all the necessary API calls to
+// delete a hosted service from Azure.
+func resourceAzureHostedServiceDelete(d *schema.ResourceData, meta interface{}) error {
+	azureClient := meta.(*AzureClient)
+	managementClient := azureClient.managementClient
+	hostedServiceClient := hostedservice.NewClient(managementClient)
+
+	log.Println("[INFO] Issuing hosted service deletion.")
+	serviceName := d.Get("service_name").(string)
+	ephemeral := d.Get("ephemeral_contents").(bool)
+	reqID, err := hostedServiceClient.DeleteHostedService(serviceName, ephemeral)
+	if err != nil {
+		return fmt.Errorf("Failed issuing hosted service deletion request: %s", err)
+	}
+
+	log.Println("[DEBUG] Awaiting confirmation on hosted service deletion.")
+	err = managementClient.WaitForOperation(reqID, nil)
+	if err != nil {
+		return fmt.Errorf("Error on hosted service deletion: %s", err)
+	}
+
+	d.SetId("")
```

Comment:
@svanharmelen pointed that out as well.
I have made all necessary adaptations for his comments on a branch where I used his stuff as a base.